From 4915745365815418ec0f21cb6adc7ed2000b6245 Mon Sep 17 00:00:00 2001
Message-Id: <4915745365815418ec0f21cb6adc7ed2000b6245.1413836944.git.chang-joon.lee@intel.com>
In-Reply-To: <5d8aa9994fff43965a6fab8c9a528b8b47a9d624.1413836944.git.chang-joon.lee@intel.com>
References: <5d8aa9994fff43965a6fab8c9a528b8b47a9d624.1413836944.git.chang-joon.lee@intel.com>
From: arsharma <ankitprasad.r.sharma@intel.com>
Date: Thu, 7 Aug 2014 13:57:30 +0530
Subject: [PATCH 53/71] MUST_REBASE [VPG]: drm/i915: Fix for Small blocking
 corruptions with VPP

Secure decode lucas tests with VPP fails with small blocking corruptions.
Using Store_Dword_Index in place of Store_Data_Immediate for flushing
pipeline.
This patch is marked MUST_REBASE because the base patch "Use deprecated
arguments to pass command buffer private data" is a MUST_REBASE

Issue: GMIN-2965
Signed-off-by: arsharma <ankitprasad.r.sharma@intel.com>
(cherry picked from commit 2c5e2bfcbdf9dd6c9860e3a5c3780cbea616c1fa)

Change-Id: Icc999e7ae27bdd67618751729dc2004339773c4b
---
 drivers/gpu/drm/i915/intel_ringbuffer.c |   18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_ringbuffer.c b/drivers/gpu/drm/i915/intel_ringbuffer.c
index b854048..d8c7c80 100644
--- a/drivers/gpu/drm/i915/intel_ringbuffer.c
+++ b/drivers/gpu/drm/i915/intel_ringbuffer.c
@@ -2121,10 +2121,11 @@ launch_cb2(struct intel_engine_cs *ring)
 		if (ret)
 			return ret;
 
-		intel_ring_emit(ring, 0x10400002); /* SDI - DW0 */
-		intel_ring_emit(ring, 0);	/* SDI - DW1 */
-		intel_ring_emit(ring, hws_pga);	/* SDI - Address */
-		intel_ring_emit(ring, 0);	/* SDI - Data */
+		intel_ring_emit(ring, MI_STORE_DWORD_INDEX);
+		intel_ring_emit(ring, I915_GEM_HWS_SCRATCH_INDEX <<
+					MI_STORE_DWORD_INDEX_SHIFT);
+		intel_ring_emit(ring, 0);
+		intel_ring_emit(ring, MI_NOOP);
 		intel_ring_advance(ring);
 	}
 
@@ -2156,10 +2157,11 @@ launch_cb2(struct intel_engine_cs *ring)
 		if (ret)
 			return ret;
 
-		intel_ring_emit(ring, 0x10400002); /* SDI - DW0 */
-		intel_ring_emit(ring, 0);	/* SDI - DW1 */
-		intel_ring_emit(ring, hws_pga);	/* SDI - Address */
-		intel_ring_emit(ring, 0);	/* SDI - Data */
+		intel_ring_emit(ring, MI_STORE_DWORD_INDEX);
+		intel_ring_emit(ring, I915_GEM_HWS_SCRATCH_INDEX <<
+					MI_STORE_DWORD_INDEX_SHIFT);
+		intel_ring_emit(ring, 0);
+		intel_ring_emit(ring, MI_NOOP);
 		intel_ring_advance(ring);
 	}
 
-- 
1.7.9.5

